### DISCUSSION TITLE
Jinja2 stub not installed

### DISCUSSION THREAD

**Discussion by patrick-5546** [pandas-dev/pandas-stubs#593] [Q&A]
When I run mypy I get the following error:
```
.tox/py310/lib/python3.10/site-packages/pandas-stubs/io/formats/style_render.pyi:13: error: Cannot find implementation or library stub for module named "jinja2"  [import]
```

I was thinking that this is because the `jinja2-stubs` package is not installed, but the weirdest part is that it doesn't happen all the time and only happens when I run it using tox. I have observed that mypy in tox passes when I first open a new terminal, but then it fails after a while. Any ideas on what might be happening, or how I could further troubleshoot this bug?

**Comment by twoertwein** [comment]
Does mypy use different config files (ignore_missing_imports) when you call it with/without tox?

**Comment by Dr-Irv** [comment]
As best as I can tell, `jinja2` since version 3.0 is fully typed, so you don't need a `jinja2-stubs` package.

This issue is probably not an issue for `pandas-stubs`.  It may be because you don't have `jinja2` installed, or because of the settings in your environment.

Also, with `mypy`, it caches information from each time, so try it with the `--no-incremental` flag.
